src/BackupChain.cs

```diff
@@ -19,26 +20,28 @@ public class BackupChain : IBackupChain
   {
     private readonly IList<BackupMetadata> _orderedBackups;
 
+    // This also handles any striped backups
     private BackupChain(IList<BackupMetadata> recentBackups)
     {
+      if (recentBackups == null || !recentBackups.Any()) {

```

Comment:
`Empty` would be the appropriate check here instead of `any`

Reply:
What if a full backup doesn't exist?  Should you throw there too? What if a log backup doesn't exist?

Reply:
Don't think there is `.Empty()` for `List` - preferred strategy by others seem to be this too:
https://stackoverflow.com/questions/24390005/checking-for-empty-or-null-liststring/43142462

I throw an exception in the function that searches for a full backup: 
https://github.com/factset/AgDatabaseMove/pull/62/files#diff-e1d6a44c77500bb8e79c7f8f9f446183e74633feb51f39573323c6345b8f6bf1R69

don't need one for `diff` or `logs` (they don't necessarily have to be in the chain for the restore to work)
can be `full -> diff -> log`, `full -> diff`, `full -> Log` or just `full`
(the order matters tho, `full` before `diff` and `diff` before any `logs` - creating some new tests that will ensure this)

Reply:
Gotcha. I think `Count >0` is slightly more readable.
Ah I think in CRDS jobs, we force a new log backup so there is always at least 1 log, but I suppose that isn't always required